### PR TITLE
Add reducers  and actions

### PR DESCRIPTION
**Pull request by abiola-arobieke**
In this milestone:

The Redux toolkit and react-redux packages were installed.
bookSlice.js and categoriesSlice.js files were added to the project.
store.js was added to the project and configured.
Reducers and actions were added to the project.

### CODE REVIEW

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hi @abiola-arobieke  

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

# Invalid Code Review Request 🔴 

You have submitted a project, that does not have the license file added, and the acknowledgment section completed 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/redux/categories/categoriesSlice.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = {
+  status: 'Under Construction',
+};
+
+const categoriesSlice = createSlice({
+  name: 'categories',
+  initialState,
+});
```
Comment (Shedrack-Sunday) [inline src/redux/categories/categoriesSlice.js]: - [ ] You're meant to create a  reducer that checks the status and always returns "Under construction" (the initial state should check to that string).
- Kindly creates 1 reducer and call it  `checkStatus`, then you can export by default `categoriesSlice.reducer` and  export  `checkStatus` as an action

## Sample code:

```
export const categoriesSlice = createSlice({
  name: "categories",
  initialState: {
    categories: [],
  },
  reducers: {
    checkStatus: (state, action) => {
      state.categories =
        action.payload === "Under construction"
          ? "Under construction"
          : state.categories
    },
  },
})

export const { checkStatus } = categoriesSlice.actions

export default categoriesSlice.reducer
```

## Instructions :
image

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hello @abiola-arobieke       👋 

# CHANGES REQUIRED ♻️

Great work on the project 👏
You've done well so far in the project,  However, there are a few you need to work on, to move to the next project. 💯

 


## Highlights   
- Good use of GitHub features ✔️   
- Good use of the rubber duck file structure 👍 

  
# Required Changes  🌵 

- [x] You made some impressive implementations which I must say is a job well done. However, there are a few  features you missed or need to implement

_Check the comments for a detailed review 

# Optional suggestions 

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by roshan-bajgain — APPROVED** [review]
Hi @abiola-arobieke, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- No linter errors. ✔️
- GitHub flow is followed. ✔️
- Well done on  implementing the changes ✔️ 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**